cli/src/test/lock.test.ts

```diff
@@ -0,0 +1,49 @@
+import anyTest, { TestInterface } from 'ava';
+import * as td from 'testdouble';
+import { Context, addFake } from './fake_lycan';
+import { createApp } from './createApp';
+import { success } from 'specshell';
+import { UnauthorizedError, NotFoundError } from '@binaris/spice-koa-server/interfaces';
+
+const test = anyTest as TestInterface<Context>;
+
+addFake(test);
+
+const makeApp = createApp(false);
+
+const anything = td.matchers.anything();
+
+test.beforeEach(async (t) => {
+  t.assert(success(await t.context.shell.run(`cd ${t.context.projectDir}`)));
+  t.log(t.context.projectDir);
+});
+
+test('lockApp lock an application by appName', async (t) => {
+  const app = makeApp();
+  td.when(t.context.lycanFake.getAppByName(anything, app.name)).thenResolve(app);
+  const result = await t.context.shell.run(`${t.context.run} lock --reason lock-my-app ${app.name}`, 'utf-8');
+  t.snapshot(result);
+  td.verify(t.context.lycanFake.lockApp(anything, app.id, 'lock-my-app'));
```

Comment:
(Not for this PR): It feels like `td.verify` is not exactly appropriate for this use - since the test would pass also when not calling `td.verfiy`. Testdouble [verify](https://github.com/testdouble/testdouble.js/blob/54c16f56871b82047d2e56e11d5c78e4b33b63d8/docs/6-verifying-invocations.md#verifying-interactions) is supposed to be used only with functions that have side-effects, and a `Promise<void>` function is almost the same as a `void` function that is needed only for it's side effects - but if it was a function of `Promise<LockAppResponse>` then we will be forced to handle it's response, and _forced_ to setup this function in `td.when`, and not be in a state where we can accidentally forget it.

Reply:
This sounds like an argument for a particular Concord style: always return a value.  Arguably the alternative is a `void`-or-`throw` style, which requires verification.  Not sure which, let's continue this.